libs/community/langchain_community/storage/redis.py

```diff
@@ -64,29 +64,28 @@ def __init__(
                 "pip install redis"
             ) from e
 
-        if client and redis_url or client and client_kwargs:
+        if client and (redis_url or client_kwargs):
             raise ValueError(
                 "Either a Redis client or a redis_url with optional client_kwargs "
                 "must be provided, but not both."
             )
 
+        if not client and not redis_url:
+            raise ValueError("Either a Redis client or a redis_url must be provided.")
+
         if client:
             if not isinstance(client, Redis):
                 raise TypeError(
                     f"Expected Redis client, got {type(client).__name__} instead."
                 )
             _client = client
         else:
-            if not redis_url:
```

Comment:
Could add this back -- it was probably here due to satisfy the linter for static typing

Reply:
I see, DONE :)